test/unit/proxy.test.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import { getProxyUrl, proxyUrlToJavaOptions } from '../../src/proxy';
+import { ScannerProperties, ScannerProperty } from '../../src/types';
+
+const MOCKED_PROPERTIES: ScannerProperties = {
+  [ScannerProperty.SonarHostUrl]: 'http://some-private-sonarqube.com',
+};
+
+describe('proxy', () => {
+  describe('getProxyUrl', () => {
+    it('should not detect proxy when proxy host is not provided', () => {
+      const properties: ScannerProperties = {
+        [ScannerProperty.SonarHostUrl]: 'http://sq.some-company.com',
+        [ScannerProperty.SonarScannerProxyPort]: '4234',
+        [ScannerProperty.SonarScannerProxyUser]: 'user',
+        [ScannerProperty.SonarScannerProxyPassword]: 'password',
+      };
+      getProxyUrl(properties);
+
+      expect(getProxyUrl(properties)).toBeUndefined();
+    });
```

Comment:
if a user provides some proxy properties but not all, should we inform the user with a warning about the missing proxy property?